packages/firestore/src/api/database.ts

```diff
@@ -1079,6 +1072,24 @@ export class DocumentSnapshot implements firestore.DocumentSnapshot {
   }
 }
 
+export class QueryDocumentSnapshot extends DocumentSnapshot
+  implements firestore.QueryDocumentSnapshot {
+  constructor(
+    firestore: Firestore,
+    key: DocumentKey,
+    document: Document,
+    fromCache: boolean
+  ) {
+    super(firestore, key, document, fromCache);
+  }
+
+  data(): firestore.DocumentData {
+    const data = super.data();
+    assert(!!data, 'Document in a QueryDocumentSnapshot should exist');
```

Comment:
nit: why not assert typeof data === 'object' ?  That's more precise and explicit than doing the truthy-cast-to-boolean thing.

Reply:
Fixed.